subprojects/ide/src/main/java/org/gradle/plugins/ide/eclipse/model/internal/SourceFoldersCreator.java

```diff
@@ -107,7 +109,8 @@ public String apply(SourceFolder sourceFolder) {
         return collector.apply(Pair.of(regularSourceFolders, dedupedExternalSourceFolders));
     }
 
-    private List<SourceFolder> projectRelativeFolders(Iterable<SourceSet> sourceSets, Function<File, String> provideRelativePath, File defaultOutputDir) {
+    private List<SourceFolder> projectRelativeFolders(Iterable<SourceSet> sourceSets, Function<File, String> provideRelativePath, File defaultOutputDir, Property<String> testSourceSetNamePatternProperty) {
```

Comment:
If you can pass the classpath into this method, you can find all the test SourceSets created by the test suite plugin like this:

```
final Set<SourceSet> testSuiteSourceSets = new HashMap<>()
JavaPluginExtension java = project.getExtensions().getByType(JavaPluginExtension.class);
TestingExtension testing = project.getExtensions().getByType(TestingExtension.class);
ExtensiblePolymorphicDomainObjectContainer<TestSuite> testSuites = testing.getSuites();
testSuites.forEach(suite -> {
    if (suite instanceof JvmTestSuite) {
        JvmTestSuite jvmTestSuite = (JvmTestSuite) suite;
        testSuiteSourceSets.add(jvmTestSuite.getSources());
    }
});
```

And then when iterating through all the SourceSets, also mark any directories for SourceSets in this collection as test for eclipse.

Reply:
Test `Configuration`s can be retrieved from these sourceSets, too.  Here are the 3 that test suites create for each `SourceSet`:

```
Configuration compileOnly = configurations.getByName(sourceSet.getCompileOnlyConfigurationName());
Configuration implementation = configurations.getByName(sourceSet.getImplementationConfigurationName());
Configuration runtimeOnly = configurations.getByName(sourceSet.getRuntimeOnlyConfigurationName());
```